Frontend Modularization: Catalog

itsmo-amzn opened the pull request:
### Feature or Bugfix
<!-- please choose -->
- Refactoring

### Detail
- Move catalog views into catalog module
- Move services that were only used by catalog into catalog module
- Move catalogs components into catalog module

### Relates
- [Frontend modularization](https://github.com/awslabs/aws-dataall/issues/398)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jorgeig-space commented on frontend/src/modules/Catalog/components/GlossarySearch.js:
```diff
@@ -319,4 +319,3 @@ GlossarySearch.propTypes = {
   setQuery: PropTypes.func.isRequired,
   matches: PropTypes.array.isRequired
 };
-export default GlossarySearch;
```
For 2.0.1: we should rename `GlossarySearch` and `GlossarySearchComponent` to be more consistent, i.e., when we read the name, we should have an idea of what each entity is: what is `GlossarySearch` vs `GlossarySearchComponent`? Why does one have a qualifier and the other doesn't?

itsmo-amzn replied:
renamed to `GlossarySearchWrapper` and `GlossarySearchUI`

jorgeig-space commented on frontend/src/modules/Catalog/components/Hit.js:
```diff
@@ -45,7 +45,7 @@ HitICon.propTypes = {
 };
 
-const Hit = ({ hit }) => {
+export const Hit = ({ hit }) => {
```
For 2.0.1: We should rename `Hit` to be more descriptive: `SearchHit`? `GlossarySearchResultHit`?

itsmo-amzn replied:
I believe `GlossarySearchResultItem` is much more descriptive.

jorgeig-space commented:
LGTM w.r.t. modularization. Left some comments for 2.0.1.

jorgeig-space commented:
@itsmo-amzn is it worth it merging #581 first and then using relative imports in this PR already?

itsmo-amzn commented:
> @itsmo-amzn is it worth it merging #587 first and then using relative imports in this PR already?

Yes. I rebased to fetch the new changes, and updated all relative imports in the Catalog module to reflect this

nikpodsh commented on frontend/src/services/graphql/index.js:
```diff
@@ -1,7 +1,6 @@
 export * from './Activity';
 export * from './AirflowCluster';
 export * from './ApiKeys';
-export * from './Catalog';
```
I wonder if we want to disable catalog in the routes when Dashboards and Datasets are inactive? 
Otherwise, it will be just empty

itsmo-amzn replied:
@jorgeig-space can you weigh-in on this?

jorgeig-space replied:
@dlpzx wdyt? I think ideally we should disable the Catalog if its functionality does not make sense when the other two modules that use it are inactive.

dlpzx replied:
For me `Datasets` and `Dashboards` should have a depends on on the `Catalog` module like it happens with `Datasets` and `Dataset_sharing`

itsmo-amzn replied:
@dlpzx can you please elaborate?

dlpzx replied:
Yes, sorry I was a bit cryptic. In an ideal world, the Catalog module should be enabled if either Datasets or Dashboards are enabled. Otherwise it should be disabled. But currently it is considered `core` in the backend modularization. @nikpodsh wdyt?
option 1: keep Catalog in core (backend) or always enabled (frontend)
option 2: modularize Catalog with Dataset and Dashboards depends on Catalog (backend) and make Catalog dependent on Datasets and Dashboards (frontend)
option 3: keep Catalog in core (backend) and make Catalog dependent on Datasets and Dashboards (frontend). We will modularize Catalog (and its infrastructure) after V2.0.0 release

dlpzx reviewed (commented):
looks good. I like the renaming and the separation of components and views. Plus the improvements in the imports

nikpodsh commented on frontend/src/routes.js:
```diff
@@ -219,10 +217,11 @@ const routes = [
-        path: 'catalog',
-        element: <Catalog />
-      },
+      config.modules.datasets.active &&
```
`&&` is wrong. The logic here: `if at least one of (dashboards, dashboards) is not present, catalog is not displayed` instead it should be `if all of them are not present, catalog is not displayed`
